### ISSUE TITLE
New Updated APKs installing previous versions

### ISSUE THREAD

**Issue by BarbaricGamester** [flutter/flutter#16597]
## General

I uploaded version 0.1.0 for my BETA crypto currency app a week ago and last night uploaded version 0.1.1 for it, the change was that previously my app supported 50 crypto currencies but in the newer version it supported 100. 

## Problem

The debugger was showing all my updated content, but when I uploaded the release apk on Playstore, it had the same content as the previous version, I tried building the apk again and testing it on my phone, it had the content from 0.1.0 when it was supposed to bue content from 0.1.1.

**Comment by chinmaygarde** [comment]
Are you sure you re-built the release version of the APK using the `--release` flag? The documentation for the release process [is here](https://flutter.io/android-release/#building-a-release-apk). It may be possible that you were testing the debug variant and had an old release build from a previous run. If you want to clean your source tree completely, you can try `git clean -fdx` (beware that this will get rid of all uncommitted changes so want to commit or stash progress before you try this) and try to build a release version again.

**Comment by BarbaricGamester** [comment]
Tried this... Still not working...

**Comment by zoechi** [comment]
I often need `flutter clean` to make the running app reflect changes.

**Comment by mit-mit** [comment]
I'm seeing the same issue, repro steps:

```
flutter create foobar
flutter build apk
edit lib/main.dart to show 2 in the titel bar
flutter run -d <iOS Simulator>
```
=> title bar shows 2

```
edit lib/main.dart to show 3 in the titel bar
flutter build apk
flutter install -d <Android device>
launch app on device
```

=> title bar should show 3, but still shows the original title from flutter create

**Comment by BarbaricGamester** [comment]
Finally, I got rid of the problem with the following steps, 

* run 'flutter clean'
* run 'flutter build apk --release'

Closing this as the original issue is resolved.

**Comment by blackstardlb** [comment]
Also ran into this problem.

**Comment by Areiser** [comment]
I had the same problem for a few days on beta branch. Flutter run updates fine, but --release always needs flutter clean.

**Comment by mit-mit** [comment]
@jcollins-g any chance you can help with this flutter tools issue?

**Comment by eseidelGoogle** [comment]
Is this the same as https://github.com/flutter/flutter/issues/16604 ? @cbracken @tvolkert

**Comment by tvolkert** [comment]
If this only happens on release builds, it's likely the problem that @cbracken fixed a few days ago with #17175.  Is anyone able to reproduce this on an updated master channel?
